src/build.py

```diff
@@ -106,6 +107,8 @@ def run_step(returncode, msg):
                 'There was a problem fetching the repository, see the above logs for details.'
             )
 
+            commit_sha = fetch_commit_sha(CLONE_DIR_PATH)
```

Comment:
It's pretty unlikely, but can we add some error handling here just in case? At a minimum, lets catch the error and throw a `StepException` so it can be handled the same way as the others?